src/shims/unix/linux/epoll.rs

```diff
@@ -168,25 +289,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     fn epoll_wait(
         &mut self,
         epfd: &OpTy<'tcx>,
-        events: &OpTy<'tcx>,
+        events_op: &OpTy<'tcx>,
         maxevents: &OpTy<'tcx>,
         timeout: &OpTy<'tcx>,
     ) -> InterpResult<'tcx, Scalar> {
         let this = self.eval_context_mut();
 
+        this.check_no_isolation(
+            "`epoll_wait`: The notification returned is nondeterministic as \
+            it depends on interpreter pointer addresses.",
+        )?;
```

Comment:
We shouldn't do this. We should instead make it deterministic. There's no fundamental reason why epoll needs to be non-deterministic, is there?